common/src/main/scala/explore/common/AsterismQueries.scala

```diff
@@ -93,13 +94,45 @@ object AsterismQueries:
       }
       AsterismGroupsWithObs(newAsterismGroups, newTargetsWithObs, newObservations)
     }
+
+    // Rebuilds asterism groups and observation list for each target, based on the
+    // observations asterisms.
+    def rebuildAsterismGroups: AsterismGroupsWithObs = {
+      val targetObservations: Map[Target.Id, SortedSet[Observation.Id]] =
+        observations.values
+          .flatMap(obs => obs.scienceTargetIds.map(targetId => targetId -> obs.id))
+          .groupMap(_._1)(_._2)
+          .view
+          .mapValues(obsIds => SortedSet.from(obsIds))
+          .toMap
+
+      val updatedTargets: TargetWithObsList =
+        targetsWithObs.map((targetId, targetWithObs) =>
+          targetId ->
+            TargetWithObs.obsIds.replace(targetObservations.get(targetId).orEmpty)(targetWithObs)
+        )
+
+      val observationTargets: AsterismGroupList =
```

Comment:
If we have this code to calculate the asterismGroups, should we even include them in the initial query?

Alternatively, I think the asterismGroups are only required on the target tab? If so, could they could just be created as needed there? Although I guess they only need to be recalculated when the observations change, not the targets. 🤔

Reply:
Yup, we could definitely compute this instead of querying the groups. Maybe we can even delete the maintenance methods (`insertObs`, `removeObsWithTargets`). I'll attempt that.